govcd/lbappprofile.go

```diff
@@ -91,92 +91,100 @@ func (eGW *EdgeGateway) ReadLBAppProfile(lbAppProfileConfig *types.LBAppProfile)
-func (eGW *EdgeGateway) ReadLBAppProfileByID(id string) (*types.LBAppProfile, error) {
-	return eGW.ReadLBAppProfile(&types.LBAppProfile{ID: id})
+// GetLbAppProfileById wraps GetLbAppProfile and needs only an ID for lookup
+func (egw *EdgeGateway) GetLbAppProfileById(id string) (*types.LbAppProfile, error) {
+	return egw.GetLbAppProfile(&types.LbAppProfile{ID: id})
 }
 
-// ReadLBAppProfileByName wraps ReadLBAppProfile and needs only a Name for lookup
-func (eGW *EdgeGateway) ReadLBAppProfileByName(name string) (*types.LBAppProfile, error) {
-	return eGW.ReadLBAppProfile(&types.LBAppProfile{Name: name})
+// GetLbAppProfileByName wraps GetLbAppProfile and needs only a Name for lookup
+func (egw *EdgeGateway) GetLbAppProfileByName(name string) (*types.LbAppProfile, error) {
+	return egw.GetLbAppProfile(&types.LbAppProfile{Name: name})
 }
 
-// UpdateLBAppProfile updates types.LBAppProfile with all fields. At least name or ID must be specified.
+// UpdateLbAppProfile updates types.LbAppProfile with all fields. At least name or ID must be specified.
 // If both - Name and ID are specified it performs a lookup by ID and returns an error if the specified name and found
 // name do not match.
-func (eGW *EdgeGateway) UpdateLBAppProfile(lbAppProfileConfig *types.LBAppProfile) (*types.LBAppProfile, error) {
-	err := validateUpdateLBAppProfile(lbAppProfileConfig)
+func (egw *EdgeGateway) UpdateLbAppProfile(lbAppProfileConfig *types.LbAppProfile) (*types.LbAppProfile, error) {
+	err := validateUpdateLbAppProfile(lbAppProfileConfig, egw)
 	if err != nil {
 		return nil, err
 	}
 
-	lbAppProfileConfig.ID, err = eGW.getLBAppProfileIDByNameID(lbAppProfileConfig.Name, lbAppProfileConfig.ID)
+	lbAppProfileConfig.ID, err = egw.getLbAppProfileIdByNameId(lbAppProfileConfig.Name, lbAppProfileConfig.ID)
 	if err != nil {
 		return nil, fmt.Errorf("cannot update load balancer application profile: %s", err)
 	}
 
-	httpPath, err := eGW.buildProxiedEdgeEndpointURL(types.LBAppProfilePath + lbAppProfileConfig.ID)
+	httpPath, err := egw.buildProxiedEdgeEndpointURL(types.LbAppProfilePath + lbAppProfileConfig.ID)
 	if err != nil {
 		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
 	}
 
 	// Result should be 204, if not we expect an error of type types.NSXError
-	_, err = eGW.client.ExecuteRequestWithCustomError(httpPath, http.MethodPut, types.AnyXMLMime,
+	_, err = egw.client.ExecuteRequestWithCustomError(httpPath, http.MethodPut, types.AnyXMLMime,
 		"error while updating load balancer application profile : %s", lbAppProfileConfig, &types.NSXError{})
 	if err != nil {
 		return nil, err
 	}
 
-	readAppProfile, err := eGW.ReadLBAppProfileByID(lbAppProfileConfig.ID)
+	readAppProfile, err := egw.GetLbAppProfileById(lbAppProfileConfig.ID)
 	if err != nil {
 		return nil, fmt.Errorf("unable to retrieve application profile with ID (%s) after update: %s",
 			readAppProfile.ID, err)
```

Comment:
If err is not nil, readAppProfile should be nil, and can't reference and ID

Reply:
Yes. Changed it.